processing/src/main/java/org/apache/druid/segment/virtual/ExpressionSelectors.java

```diff
@@ -492,11 +492,26 @@ static Supplier<Object> supplierFromObjectSelector(
           return val;
         }
       };
+    } else if (clazz.isAssignableFrom(Object[].class)) {
+      return () -> {
+        final Object val = selector.getObject();
+        if (val != null) {
+          NonnullPair<ExpressionType, Object[]> coerced = ExprEval.coerceListToArray(
+              Arrays.asList((Object[]) val),
+              homogenizeMultiValue
```

Comment:
wait, actual arrays should _never_ be coerced, this is strictly for multi-value strings, why do we need to coerce here?

Reply:
Thanks @clintropolis, Understood. changed it to pass through directly for Object arrays. I checked on few scenarios and they worked fine. Would this not create any problem if the underlying Object is not String or Numeric? I tried Array<Array<String>> and that worked.